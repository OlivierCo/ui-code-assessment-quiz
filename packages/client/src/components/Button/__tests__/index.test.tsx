import React from 'react'
import { shallow, mount } from 'enzyme'
import { Button } from '../index'

describe('Button', () => {
  it('should be defined', () => {
    expect(Button).toBeDefined()
  })

  it('should render correctly', () => {
    const wrapper = shallow(
      <Button onClick={() => console.log('test')}>my button</Button>,
    )
    expect(wrapper).toMatchSnapshot()
  })

  it('should call mock function when button is clicked', () => {
    const mockFn = jest.fn()
    const wrapper = shallow(<Button onClick={mockFn}>my button</Button>)

    wrapper.simulate('click')
    expect(mockFn).toHaveBeenCalled()
  })

  it('should be disabled', () => {
    const mockOnClick = jest.fn()
    const wrapper = mount(
      <Button onClick={mockOnClick} disabled>
        my button
      </Button>,
    )

    wrapper.simulate('click')
    expect(mockOnClick).toHaveBeenCalledTimes(0)
  })
})
