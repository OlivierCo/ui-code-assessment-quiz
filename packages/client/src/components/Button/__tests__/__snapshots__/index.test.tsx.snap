// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button should render correctly 1`] = `
<EmotionCssPropInternal
  __EMOTION_LABEL_PLEASE_DO_NOT_USE__="Button"
  __EMOTION_TYPE_PLEASE_DO_NOT_USE__="button"
  css={
    Object {
      "map": undefined,
      "name": "1f9n0ck",
      "next": undefined,
      "styles": "width:100%;padding:15px 0px;cursor:pointer;color:#fff;outline:none;border:none;background-color:#4f5df7;margin-top:20px;font-size:18px;&:hover{background:rgba(42, 70, 96, 0.8);}&:active{background:rgba(42, 70, 96, 0.9);}&:disabled{background:#cccccc;color: #666666;cursor:not-allowed;}",
      "toString": [Function],
    }
  }
  onClick={[Function]}
  type="button"
>
  my button
</EmotionCssPropInternal>
`;
